lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkgateway.py

```diff
@@ -0,0 +1,431 @@
+        self.gateway_default_site = None
+        self.sku = None
+        self.vpn_client_configuration = None
+        self.bgp_settings = None
+        self.active_active = None
+
+        self.results = dict(
+            changed=False,
+            state=dict()
+        )
+
+        super(AzureRMVirtualNetworkGateway, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                           supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        changed = False
+        results = dict()
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        self.virtual_network = self.parse_resource_to_dict(self.virtual_network)
+        resource_group = self.get_resource_group(self.resource_group)
+
+        try:
+            vgw = self.network_client.virtual_network_gateways.get(self.resource_group, self.name)
+            self.check_provisioning_state(vgw, self.state)
+            results = vgw_to_dict(vgw)
+            if self.state == 'present':
+                update_tags, results['tags'] = self.update_tags(results['tags'])
+                if update_tags:
+                    changed = True
+                    fh.write('changed in tags\n')
+                sku = dict(name=self.sku, tier=self.sku)
+                if sku != results['sku']:
+                    changed = True
+                if self.enable_bgp != results['enable_bgp']:
+                    changed = True
+                if self.bgp_settings and self.bgp_settings['asn'] != results['bgp_settings']['asn']:
+                    changed = True
+                if self.active_active != results['active_active']:
+                    changed = True
+            elif self.state == 'absent':
+                self.log("CHANGED: vnet exists but requested state is 'absent'")
+                changed = True
+        except CloudError:
+            if self.state == 'present':
+                self.log("CHANGED: VPN Gateway {0} does not exist but requested state is 'present'".format(self.name))
+                changed = True
+
+        self.results['changed'] = changed
+        self.results['state'] = results
+
+        if self.check_mode:
+            return self.results
+        if changed:
+            if self.state == 'present':
+                if not self.sku:
+                    self.fail('Parameter error: sku is required when creating a vpn gateway')
+                if not self.ip_configurations:
+                    self.fail('Parameter error: ip_configurations required when creating a vpn gateway')
+                subnet = self.network_models.SubResource(
+                    '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Network/virtualNetworks/{2}/subnets/GatewaySubnet'.format(
```

Comment:
@Madhura-CSI i have just started working on this pr, and I have a quick question.
why is GatewaySubned hardcoded here. shouldnt' we refer any subnet?
any special reason?

Reply:
@zikalino,  'GatewaySubnet' < this name is mandatory for the subnet where we deploy the virtual network gateway virtual machines. AFAIK, we cannot change this name. If we ever do that, network gateway creation will fail at the end. Please confirm this behavior.

Reply:
@Madhura-CSI thanks! I will create PR to your fork soon :-)